src/views/TopologyDisplay.vue

```diff
@@ -0,0 +1,340 @@
+      if (result.config === undefined) continue;
+      const display: DisplayConfig[] = [];
+      const title = result.config.timeSeriesDisplay.title ?? ''
+      result.config.timeSeriesDisplay.subplots?.forEach((subPlot, index) => {
+        display.push({
+          id: `${title}-${index}`,
+          types: [DisplayType.TimeSeriesChart, DisplayType.TimeSeriesTable],
+          class: 'single',
+          title: title,
+          config: timeSeriesDisplayToChartConfig(subPlot, title)
+        })
+      })
+      this.allDisplays.push(display);
+      this.requests.push(result.requests);
+      this.plots.push(title)
+    }
+    this.displays = this.allDisplays.length > 0 ? this.allDisplays[0] : [];
+    await this.loadTimeSeries(0);
+  }
+
+  createTopologyMap(nodes: TopologyNode[] | undefined, topologyMap: Map<string, TopologyNode>) {
+    if (nodes === undefined) return undefined;
+    for (const node of nodes) {
+      topologyMap.set(node.id, node)
+      this.createTopologyMap(node.topologyNodes, topologyMap)
+    }
+  }
+
+  private async loadTimeSeries(index: number) {
+    await this.updateTimeSeries(this.requests[index], {startTime: this.startTime, endTime: this.endTime, thinning: true})
+  }
+
+
+  async loadNodes(topologyNodes: TopologyNodeResponse): Promise<ColumnItem[]> {
+    const recursiveUpdateNode = (nodes: TopologyNode[]) => {
+      return nodes.filter(node => this.topologyNodeIsVisible(node)).map((node) => {
+        const result: ColumnItem = {
+          id: node.id,
+          name: node.name,
+          icon: this.getIcon(node)
+        }
+        if (node.topologyNodes) {
+          result.children = recursiveUpdateNode(node.topologyNodes)
+        }
+        if (node.url !== undefined && node.mainPanel !== TopologyDisplay.TIME_SERIES_DIALOG_PANEL) {
+          result.href = node.url
+          result.target = node.url
+        } else {
+          result.wmsLayerId = node.gridDisplaySelection !== undefined ? node.gridDisplaySelection.plotId : undefined
+          result.filterIds = node.filterIds ?? []
+          result.to = {
+            name: 'TopologyDisplay',
+            params: {
+              nodeId: node.id
+            }
+          }
+        }
+        return result
+      })
+    }
+    return [
+      {
+        id: 'root',
+        name: 'Topology',
+        children: recursiveUpdateNode(topologyNodes.topologyNodes)
+      }
+    ]
+  }
+
+  getTopologyTabs(topologyNodeResponse: TopologyNodeResponse): TopologyDisplayTab[] {
+    const panels = []
+    const hasGrid = this.hasApplicableNodes(topologyNodeResponse.topologyNodes, node => node.gridDisplaySelection !== undefined || node.filterIds != undefined)
+    const hasTimeSeriesDisplay = this.hasApplicableNodes(topologyNodeResponse.topologyNodes, node => node.displayGroups !== undefined)
+    if (hasGrid || hasTimeSeriesDisplay) {
+      panels.push({id: "spatialDisplay", title: "Time Series Display"} as TopologyDisplayTab)
```

Comment:
In deze gevallen vind ik het wel mooi om het array waar je naar pusht al te type-annotaten, dan hoef je hier niet meer te casten, maar dat is misschien een kwestie van smaak